### PR TITLE
 Fixed some potential problems

### PR DESCRIPTION
**Pull request by EnesOkutan**
Fixed some problems that may arise in the choice of namespace and module.

## Description
* Adding modules with the same name is blocked.
* Removed letter sensitivity for namespace.

## Motivation and Context
* If a module with the same name is added, the previous module cannot be accessed. This is a problem.
* The namespace selection in the command interface is case insensitive. Accessing problems may occur if names of the same name but different letter sizes are added while adding. This is a problem.

For example: 

namespace: kernel
namespace: Kernel

switch kernel => Access problem (because case insensitive)

* Users should not use the namespace that the operating system uses. If used, it cannot access the added module. This is a problem.

## How Has This Been Tested?
Tested on Arch Linux

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Review by bufgix — APPROVED** [review]
:+1:

File: core/cli/commands.py [hunk 1]
```diff
@@ -380,3 +382,14 @@ def validate(self, document):
...
                     message='Enter a valid module name',
                     cursor_position=len(document.text))
+            elif document.text.lower() in [module.lower() for module in self.modules]:
+                raise ValidationError(
+                    message="Try a different name, this module name is available",
+                    cursor_position=len(document.text))
+
+class NamespaceValidation(Validator):
+    def validate(self, document):
+        if document.text == ("kernel" if check_os() else "firewall"):
```
Comment (orhun) [inline core/cli/commands.py]: It doesn't seem convenient to me. We have to add bunch of new controls as we add new namespaces. So it should be replaced with something more appropriate for the context.

File: core/cli/commands.py [hunk 2]
```diff
@@ -217,17 +215,21 @@ def save_mod_json(data):
...
+                        'validate': NamespaceValidation
+                    }]
+                    mod_namespace = prompt(mod_question)
                 try:
-                    mod_ns = mod_details['mod_ns']
+                    mod_ns = mod_namespace['mod_ns']
                 except:
                     pass
+                #The modules in the selected namespace are static defined to the ModuleNameValidation class.
+                ModuleNameValidation.modules = self.modules[mod_ns].keys() if mod_ns in self.modules.keys() else []
```
Comment (orhun) [inline core/cli/commands.py:230]: Can you explain more about this line please?

**Comment by orhun** [comment]
Some points caught my eye. See the comments +

**Review by bufgix — APPROVED** [review]

**Review by furk4nyuksel — APPROVED** [review]

File: core/cli/commands.py (hunk 1 above)
Comment (EnesOkutan) [reply core/cli/commands.py]: Replaced with a more appropriate structure.

File: core/cli/commands.py (hunk 2 above)
Comment (EnesOkutan) [reply core/cli/commands.py:230]: Added a more descriptive comment line.

**Comment by EnesOkutan** [comment]
> Some points caught my eye. See the comments +

Your comment has been reviewed, and some points have been changed.

**Review by afk — APPROVED** [review]

File: core/cli/commands.py (hunk 2 above)
Comment (orhun) [reply core/cli/commands.py:230]: Actually, line below that.

**Pull request merged**